### DISCUSSION TITLE
Add header in context request sent to getServerSideProps

### DISCUSSION THREAD

**Discussion by Chanizo** [vercel/next.js#69872] [Help]
### Summary

Hi,
It's not clear for us how to add a header in a request received by getServerSideProps with context object.
We want to add a correlation-id in header. Maybe I'm misunderstanding how getServerSideProps work. 

Context :
We have a Nextjs app with front and server. We generate a correlation-id at client side, using local storage, and adding it in each axios request to the server with :
```
if (typeof window !== 'undefined') {
  const correlationId = getCorrelationId();
  axios.interceptors.request.use((config) => {
    if (config.headers) {
      config.headers['x-correlation-id'] = correlationId;
    }
    return config;
  }, (error) => {
    return Promise.reject(error);
  });
}```

We want to add it in the same manner in the requests received by getServerSideProps, to be able to log it to follow user errors

Thanks for help

### Additional information

_No response_

### Example

_No response_

**Comment by icyJoseph** [comment]
Doesn't the context object in GSSP, have the response object as well? https://nextjs.org/docs/pages/api-reference/functions/get-server-side-props#context-parameter

So you could attach the correlation-id to the `context.res` object. Though I am not sure about what flow you looking for here.

**Reply by Chanizo** [reply]
In fact our goal is not to add it in the response, but get it from the request in the getServerSideProps. We want to track error in getServerSideProps code, and link the log to the user with our existing correlation id generated from client side

**Reply by icyJoseph** [reply]
Who would put it there? In the request headers?  The browser ain't gonna do that 🤔

**Reply by Chanizo** [reply]
That's my question, that if it was possible ? 😉 
At client side we add the axios interceptor, like written in my first post, in the _app.tsx, correlation-id is saved in localStorage at first launch. That permits us to follow the user requests to our server. That's why I would like to do same thing when getServerSideProps is called.
But like I said, I am surely misunderstanding, it's not very clear to me how getServerSideProps works...

**Reply by icyJoseph** [reply]
Right, GSSP roughly works in two modes.

- A user navigates directly into the page that's using GSSP, and the entire HTML document is prepared for that user, by executing GSSP to get props and then rendering your app tree 

- A user is already at your application, and navigates, using next/link or router, to your page that's using GSSP, now JSON is collected from running GSSP, and sent to the browser to render the page, on the client 

Let's not consider hydration here.

So, a header won't really cut it here, maybe a cookie is what you need?

**Reply by Chanizo** [reply]
Thank you for taking the time to explain !

If indeed we have no access to how json is requested to GSSP, a cookie would be a good solution
